aptos-move/mvhashmap/src/versioned_data.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::{Incarnation, MVDataError, MVDataOutput, TxnIndex, Version};
+use aptos_aggregator::{
+    delta_change_set::{deserialize, DeltaOp},
+    transaction::AggregatorValue,
+};
+use aptos_infallible::Mutex;
+use aptos_types::write_set::TransactionWrite;
+use crossbeam::utils::CachePadded;
+use dashmap::DashMap;
+use std::{
+    collections::btree_map::BTreeMap,
+    hash::Hash,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+
+const FLAG_DONE: usize = 0;
+const FLAG_ESTIMATE: usize = 1;
+
+/// Every entry in shared multi-version data-structure has an "estimate" flag
+/// and some content.
+struct Entry<V> {
+    /// Used to mark the entry as a "write estimate". Even though the entry
+    /// lives inside the DashMap and the entry access will have barriers, we
+    /// still make the flag Atomic to provide acq/rel semantics on its own.
+    flag: AtomicUsize,
```

Comment:
rename to "estimation"?

Reply:
it had _DONE and _ESTIMATE values so can't just rename.
can make an atomicboolean and then it can be "estimation_flag" or smt. Do you prefer that?